src/targets/gpu/fuse_mlir.cpp

```diff
@@ -347,65 +376,42 @@ struct find_mlir_standalone_dot_op : find_mlir_standalone_op
  * intended to be primarily used by rocMLIR developers.
  */
 MIGRAPHX_DECLARE_ENV_VAR(MIGRAPHX_MLIR_USE_SPECIFIC_OPS);
-bool is_self_decide() { return string_value_of(MIGRAPHX_MLIR_USE_SPECIFIC_OPS{}, "").empty(); }
 
-bool is_requested(std::string_view option)
+bool is_requested(std::string_view option, bool fallback = false)
 {
-    assert(not is_self_decide());
     auto string_value  = string_value_of(MIGRAPHX_MLIR_USE_SPECIFIC_OPS{}, "");
+    if(string_value.empty())
+        return fallback;
     const auto options = split_string(string_value, ',');
     return contains(options, option);
 }
-
-bool is_enabled(std::string_view op_name, context* ctx)
-{
-    if(is_self_decide())
-    {
-        if(op_name == "fused")
-        {
-            return true;
-        }
-        else if(op_name == "convolution" or op_name == "quant_convolution")
-        {
-            if(ctx == nullptr)
-            {
-                return false;
-            }
-            else
-            {
-                const auto& device = ctx->get_current_device();
-                const std::string navi_family{"gfx110"};
-                return starts_with(device.get_gfx_name(), navi_family);
-            }
-        }
-        else
-        {
-            return false;
-        }
-    }
-    return is_requested(op_name);
-}
 } // namespace
 
 #endif // MIGRAPHX_MLIR
 
 void fuse_mlir::apply(module_pass_manager& mpm) const
 {
 #ifdef MIGRAPHX_MLIR
-    if(is_enabled("fused", this->ctx))
-    {
-        match::find_matches(mpm, find_mlir_fused_ops{});
-    }
+    const auto& device = ctx->get_current_device();
+    const std::string navi_family{"gfx110"};
+    const bool is_navi = starts_with(device.get_gfx_name(), navi_family);
 
-    if(is_enabled("convolution", this->ctx))
-    {
-        match::find_matches(mpm, find_mlir_standalone_convolution_op{});
-    }
+    auto get_mode = [&](std::string_view option, mlir_mode m1, mlir_mode m2 = mlir_mode::fast) {
+        if(is_requested(option))
+            return mlir_mode::all;
+        if(is_navi)
+            return mlir_mode::all;
+        return std::max(m1, m2);
+    };
 
-    if(is_enabled("dot", this->ctx))
-    {
-        match::find_matches(mpm, find_mlir_standalone_dot_op{});
-    }
+    mlir_mode mode = enabled(MIGRAPHX_ENABLE_MLIR{}) ? mlir_mode::fast : mlir_mode::none;
```

Comment:
Hold on, since we now have `MIGRAPHX_DISABLE_MLIR`, shouldn't this be keying off of `_DISABLE_MLIR` and not `_ENABLE_MLIR`?

Reply:
No, DISABLE_MLIR will disable MLIR completely(and its already handled in target.cpp), whereas ENABLE_MLIR will enable it for gemm fusions(because it is not enabled by default because its not always faster).

Reply:
Ah. So maybe we should rename it to something like `MIGRAPHX_ENABLE_MLIR_GEMM_FUSION` because it's a very confusing variable name